hbase-server/src/test/java/org/apache/hadoop/hbase/quotas/TestSpaceQuotasAtNamespaceLevel.java

```diff
@@ -0,0 +1,143 @@
+      HBaseClassTestRule.forClass(TestSpaceQuotasAtNamespaceLevel.class);
+
+  private static final Logger LOG = LoggerFactory.getLogger(TestSpaceQuotasAtNamespaceLevel.class);
+  private static final HBaseTestingUtility TEST_UTIL = new HBaseTestingUtility();
+
+  @Rule
+  public TestName testName = new TestName();
+  private SpaceQuotaHelperForTests helper;
+
+  @BeforeClass
+  public static void setUp() throws Exception {
+    Configuration conf = TEST_UTIL.getConfiguration();
+    SpaceQuotaHelperForTests.updateConfigForQuotas(conf);
+    TEST_UTIL.startMiniCluster(1);
+  }
+
+  @AfterClass
+  public static void tearDown() throws Exception {
+    TEST_UTIL.shutdownMiniCluster();
+  }
+
+  @Before
+  public void removeAllQuotas() throws Exception {
+    helper = new SpaceQuotaHelperForTests(TEST_UTIL, testName, new AtomicLong(0));
+    helper.removeAllQuotas();
+  }
+
+  @After
+  public void removeQuotas() throws Exception {
+    helper.removeAllQuotas();
+  }
+
+  @Test
+  public void testSetNamespaceQuotaAndRemove() throws Exception {
+    NamespaceDescriptor nd = helper.createNamespace();
+    TableName table = helper.createTableInNamespace(nd);
+
+    // Set quota on namespace.
+    helper.setQuotaLimitNamespace(nd.getName(), SpaceViolationPolicy.NO_WRITES, 2L);
+
+    // Sufficient time for all the chores to run
+    Thread.sleep(5000);
+
+    // Get Current Snapshot from 'hbase:quota'
+    Map<TableName, SpaceQuotaSnapshot> snapshotMap =
+        QuotaTableUtil.getSnapshots(TEST_UTIL.getConnection());
+
+    // After setting quota on namespace, 'hbase:quota' should have some entries present.
+    Assert.assertEquals(1, snapshotMap.size());
+
+    helper.removeQuotaFromNamespace(nd.getName());
+
+    // Get Current Snapshot from 'hbase:quota'
+    snapshotMap = QuotaTableUtil.getSnapshots(TEST_UTIL.getConnection());
+
+    // After removing quota on namespace, 'hbase:quota' should not have any entry present.
+    Assert.assertEquals(0, snapshotMap.size());
+
+    // drop table and namespace.
+    TEST_UTIL.getAdmin().disableTable(table);
+    TEST_UTIL.getAdmin().deleteTable(table);
+    TEST_UTIL.getAdmin().deleteNamespace(nd.getName());
+  }
+
+  @Test
+  public void testDropTableInNamespaceQuota() throws Exception {
+    NamespaceDescriptor nd = helper.createNamespace();
+    TableName table = helper.createTableInNamespace(nd);
+
+    // Set quota on namespace.
+    helper.setQuotaLimitNamespace(nd.getName(), SpaceViolationPolicy.NO_WRITES, 2L);
+
+    // write some data.
+    helper.writeData(table,SpaceQuotaHelperForTests.ONE_KILOBYTE);
+
+    // Sufficient time for all the chores to run
+    Thread.sleep(5000);
+
+    // Get Current Snapshot from 'hbase:quota'
+    Map<TableName, SpaceQuotaSnapshot> snapshotMap =
+        QuotaTableUtil.getSnapshots(TEST_UTIL.getConnection());
+
+    // Table before drop should have entry in 'hbase:quota'
+    Assert.assertTrue(snapshotMap.containsKey(table));
```

Comment:
Please validate that the `SpaceQuotaSnapshot` is sane (e.g. is a space quota, is the limit you set)